File: source/3090.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        if s == "":
            return 0
        left, right = 0, 0
        appear_cnts = {}
        ans = 0
        while right < len(s):
            if s[right] in appear_cnts and appear_cnts[s[right]] == 2:
                ans = max(ans, right - left)
                while appear_cnts[s[right]] == 2:
                    appear_cnts[s[left]] -= 1
                    left += 1
            else:
                if s[right] in appear_cnts:
                    appear_cnts[s[right]] += 1
                else:
                    appear_cnts[s[right]] = 1
            right += 1
        ans = max(ans, right - left)
        return ans

File: source/3090.cpp
#include <algorithm>
#include <array>
#include <print>
#include <string>

class Solution {
 public:
  int maximumLengthSubstring(std::string s) {
    int n = s.size();
    if (n == 0) {
      return 0;
    }

    std::array<int, 26> char_cnts;
    char_cnts.fill(0);
    int ans = 0;

    int left = 0, right = 0;
    while (right < s.size()) {
      if (char_cnts[s[right] - 'a'] == 2) {
        ans = std::max(ans, right - left);
        while (char_cnts[s[right] - 'a'] == 2) {
          --char_cnts[s[left] - 'a'];
          ++left;
        }
      }

      ++char_cnts[s[right] - 'a'];
      ++right;
    }

    ans = std::max(ans, right - left);
    return ans;
  }
};

int main(int /*argc*/, char* /*arcv*/[]) {
  std::println("{}", Solution().maximumLengthSubstring("bcbbbcba"));
  return 0;
}